### ISSUE TITLE
scverse certiicates

### ISSUE THREAD

**Issue by aterrel** [numfocus/infrastructure#40]
### Project

scverse

### Observations

See comments from [ivirshup](https://github.com/ivirshup) at: https://github.com/numfocus/infrastructure/issues/34#issuecomment-2146306762

Looks like the github certificate expired and switching to ssl termination broke muon.scverse.com. Need to debug options.

**Comment by aterrel** [comment]
@ivirshup please me the email list of folks who should access your cloudflare instance and I'll try to set it up for you to see yourself.

**Comment by aterrel** [comment]
It appears switching to full mode rather than strict works for both muon.scverse.org and scverse.org


<img width="1033" alt="Screenshot 2024-06-04 at 9 06 03 AM" src="https://github.com/numfocus/infrastructure/assets/30583/ce7b8063-39e8-4922-af6f-70879d689010">

**Comment by martey** [comment]
This is my fault. When I set up the scverse.org DNS records in CloudFlare, I set up the Github Pages records (scverse.org, www.scverse.org, and muon.scverse.org) to be proxied through CloudFlare. Because Github Pages had previously generated a SSL certificate through Let's Encrypt, everything seemed to be working when I changed the encryption mode to "Full (Strict)". This mode does not allow Github Pages to renew the certificate, though, so this issue started when the Github Pages certificate expired.

Andy's change of the encryption mode to "Full" does fix the problem, but at some point in the future we should disable proxying on the records that point to Github Pages and tell Github to try and renew the certificates again. This would allow switching back to "Full (strict)", which is more secure.

**Comment by aterrel** [comment]
@martey so you think Github is not renewing the certificate correctly? Perhaps we could set up a test for this on another repository.

**Comment by ivirshup** [comment]
@aterrel could you add `ivirshup` and `bredikhin.daniel` at gmail to this? Thanks!

**Comment by martey** [comment]
> so you think Github is not renewing the certificate correctly? Perhaps we could set up a test for this on another repository.

Yes. https://community.cloudflare.com/t/github-pages-keep-saying-it-cant-enforce-https/397570/2 and a few other support threads I found suggest that Github won't provision/renew certificates if the domain's DNS records do not resolve to Github IP addresses (which is the case when DNS records are being proxied through Cloudflare).

We switched scverse.org to Cloudflare in the middle of last month (around May 19th) and  LetsEncrypt certificates are generally able to renewed a month before expiration, so the certificate's expiration suggests that Github might be waiting until just before expiration to try and renew. I'm also not sure why Github doesn't warn you when there are issues renewing a certificate.

**Comment by aterrel** [comment]
Is there anything else that needs to be done for this ticket?

**Comment by aterrel** [comment]
Closing due to lack of response. Please reopen if needed.
